src/components/modals/modification/ApproximateByRatios.vue

```diff
@@ -0,0 +1,286 @@
+
+// Make the number input skip to the next prime when incremented
+function modifyPrimeLimit(event: Event) {
+  const newValue = parseInt((event.target as HTMLInputElement).value);
+  if (isNaN(newValue)) {
+    return;
+  }
+  if (PRIMES.includes(newValue)) {
+    primeLimit.value = newValue;
+  }
+  const index = PRIMES.indexOf(primeLimit.value);
+  if (newValue < primeLimit.value && index > 1) {
+    primeLimit.value = PRIMES[index - 1];
+  }
+  if (newValue > primeLimit.value && index < PRIMES.length - 1) {
+    primeLimit.value = PRIMES[index + 1];
+  }
+}
+
+function primeLimitString(fraction: Fraction) {
+  const limit = getPrimeLimit(fraction, 97);
+  if (limit < Infinity) {
+    return `${limit}-limit`;
+  }
+  return ">97-limit";
+}
+
+type Approximation = {
+  fraction: Fraction;
+  error: number;
+  limit: string;
+};
+
+const approximationsWithErrorsAndLimits = computed<Approximation[]>(() => {
+  const selected = props.scale.getMonzo(degree.value);
+  const selectedCents = selected.totalCents();
+  if (method.value === "convergents") {
+    const approximations = getConvergents(
+      selected.valueOf(),
+      undefined,
+      2 * MAX_LENGTH, // Extra length buffer
+      includeSemiconvergents.value,
+      includeNonMonotonic.value
+    );
+    const result: Approximation[] = [];
+    approximations.forEach((fraction) => {
+      const error = Math.abs(selectedCents - valueToCents(fraction.valueOf()));
+      if (error <= maxError.value) {
+        const limit = primeLimitString(fraction);
+        result.push({ fraction, error, limit });
+      }
+    });
+    return result.slice(0, MAX_LENGTH);
+  } else if (method.value === "odd") {
+    const approximationsAndErrors = approximateOddLimitWithErrors(
+      selectedCents,
+      oddLimit.value
+    );
+    const result: Approximation[] = [];
+    approximationsAndErrors.forEach(([fraction, error]) => {
+      if (error <= maxError.value) {
+        const limit = primeLimitString(fraction);
+        result.push({ fraction, error, limit });
+      }
+    });
+    if (!result.length) {
+      const [fraction, error] = approximationsAndErrors[0];
+      return [{ fraction, error, limit: primeLimitString(fraction) }];
+    }
+    return result.slice(0, MAX_LENGTH);
+  }
+  const approximationsAndErrors = approximatePrimeLimitWithErrors(
+    selectedCents,
+    PRIMES.indexOf(primeLimit.value),
+    maxExponent.value,
+    Math.min(600, maxError.value),
+    MAX_LENGTH
+  );
+  if (!approximationsAndErrors.length) {
+    const [fraction, error] = approximatePrimeLimitWithErrors(
+      selectedCents,
+      PRIMES.indexOf(primeLimit.value),
+      maxExponent.value,
+      undefined,
+      1
+    )[0];
+    return [{ fraction, error, limit: primeLimitString(fraction) }];
+  }
+  return approximationsAndErrors.map(([fraction, error]) => ({
+    fraction,
+    error,
+    limit: primeLimitString(fraction),
+  }));
+});
+
+watch(approximationsWithErrorsAndLimits, (newValue) => {
```

Comment:
These watchers seem to trigger even when the modal is not loaded. That's a problem in terms of performance...